dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
What is the sense in allowing `dash::Atomic` on any type that will fail to perform correct atomic operation? 

The name of the trait is `dash::is_atomic_compatible` and  neither a `struct` nor `double` are *compatible* because the operations on the are not well defined.